mmcv/transforms/processing.py

```diff
@@ -1127,6 +1130,32 @@ def flip_keypoints(self, keypoints: np.ndarray, img_shape: Tuple[int, int],
         flipped = np.concatenate([keypoints, meta_info], axis=-1)
         return flipped
 
+    def flip_seg_map(self, seg_map: dict, direction: str) -> np.ndarray:
+        """Flip segmentation map horizontally, vertically or diagonally.
+
+        Args:
+            seg_map (numpy.ndarray): segmentation map, shape (H, W).
+            direction (str): Flip direction. Options are 'horizontal',
+                'vertical'.
+        Returns:
+            numpy.ndarray: Flipped keypoints.
+        """
+        seg_map = mmcv.imflip(seg_map, direction=direction)
+        if self.swap_label_pairs is not None:
+            # to handle datasets with left/right annotations
+            # like 'Left-arm' and 'Right-arm' in LIP dataset
+            # Modified from https://github.com/openseg-group/openseg.pytorch/blob/master/lib/datasets/tools/cv2_aug_transforms.py # noqa:E501
+            # Licensed under MIT license
+            temp = seg_map.copy()
+            assert isinstance(self.swap_label_pairs, (tuple, list))
+            for pair in self.swap_label_pairs:
+                assert isinstance(pair, (tuple, list)) and len(pair) == 2, \
+                    'swap_label_pairs must be a sequence with pair, but got ' \
+                    f'{self.swap_label_pairs}.'
+                seg_map[temp == pair[0]] = pair[1]
+                seg_map[temp == pair[1]] = pair[0]
```

Comment:
```suggestion
            assert isinstance(self.swap_label_pairs, (tuple, list))
            for pair in self.swap_label_pairs:
                assert isinstance(pair, (tuple, list)) and len(pair) == 2, \
                    'swap_label_pairs must be a sequence with pair, but got ' \
                    f'{self.swap_label_pairs}.'
                seg_map[seg_map == pair] = pair[::-1]
```

Reply:
NumPy boolean array indexing assignment cannot assign 2 input values to the 0 output values where the mask is true